clients/src/main/java/org/apache/kafka/common/requests/VoteResponse.java

```diff
@@ -47,13 +62,42 @@ protected Struct toStruct(short version) {
         return data.toStruct(version);
     }
 
+    public static VoteResponseData singletonResponse(Errors topLevelError,
+                                                     TopicPartition topicPartition,
+                                                     Errors partitionLevelError,
+                                                     int leaderEpoch,
+                                                     int leaderId,
+                                                     boolean voteGranted) {
+        return new VoteResponseData()
+            .setErrorCode(topLevelError.code())
+            .setTopics(Collections.singletonList(
+                new VoteResponseData.VoteTopicResponse()
+                    .setTopicName(topicPartition.topic())
+                    .setPartitions(Collections.singletonList(
+                        new VoteResponseData.VotePartitionResponse()
+                            .setErrorCode(partitionLevelError.code())
+                            .setLeaderId(leaderId)
+                            .setLeaderEpoch(leaderEpoch)
+                            .setVoteGranted(voteGranted)))));
+    }
+
     @Override
     public Map<Errors, Integer> errorCounts() {
-        return Collections.singletonMap(Errors.forCode(data.errorCode()), 1);
+        Map<Errors, Integer> errors = new HashMap<>();
+        Errors topLevelError = Errors.forCode(data.errorCode());
+        if (topLevelError != Errors.NONE) {
+            errors.put(topLevelError, 1);
```

Comment:
Seems like we handle the top-level error code a bit inconsistently throughout the codebase. I am wondering if we could use the same approach as in `LeaderAndIsrResponse`. If the top-level error is not `NONE`, then we set the count for that error to the number of partitions. I think the benefit of this approach is that it normalizes the error counts.